### PR TITLE
ref(js): Move jest instrumentation to library

### PR DESCRIPTION
**Pull request by billyvg**
This moves our jest instrumentation to use an external library: https://www.npmjs.com/package/jest-sentry-environment

The library is mostly the same, but we did add some "missing" instrumentation (`getVmContext`) -- although it's tracked as a single span, it's actually many spans, but it pollutes the UI a bit so we treat it as a single span for now.

image

### CODE REVIEW

**Review by matejminar — APPROVED** [review]
👏

File: jest.config.ts [hunk 1]
```diff
@@ -96,8 +97,20 @@ const config: Config.InitialOptions = {
...
+    sentryConfig: {
+      init: {
+        dsn: 'https://3fe1dce93e3a4267979ebad67f3de327@sentry.io/4857230',
+        environment: !!process.env.CI ? 'ci' : 'local',
+        tracesSampleRate: 1.0,
+      },
+      transactionOptions: {
+        tags: {
+          branch: process.env.GITHUB_REF,
+          commit: process.env.GITHUB_SHA,
```
Comment (billyvg) [inline jest.config.ts:109]: For PR's this is the merge commit, so I may change these up

File: package.json [hunk 2]
```diff
@@ -169,6 +169,7 @@
     "jest-circus": "27.0.6",
     "jest-fetch-mock": "^3.0.3",
     "jest-junit": "^9.0.0",
+    "jest-sentry-environment": "1.1.4",
```
Comment (EvanPurkhiser) [inline package.json:172]: I wonder if we can write this in typescript now that it's in a npm module that can have a build 👀

Comment (EvanPurkhiser) [reply package.json:172]: Not that we couldn't before I guess 😆

**Review by evanpurkhiser — APPROVED** [review]
nice stuff billy

File: package.json (hunk 2 above)
Comment (billyvg) [reply package.json:172]: We could, I'm not going to do that - I'd prefer (if this library does gain traction), that the SDK team takes it into their monorepo

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**